crates/astria-core/src/execution/v1alpha2/mod.rs

```diff
@@ -81,8 +90,11 @@ impl Protobuf for GenesisInfo {
             sequencer_genesis_block_height,
             celestia_block_variance,
         } = raw;
-        let rollup_id =
-            RollupId::try_from_slice(rollup_id).map_err(Self::Error::incorrect_rollup_id_length)?;
+        let Some(rollup_id) = rollup_id else {
+            return Err(Self::Error::no_rollup_id());
+        };
+        let rollup_id = RollupId::try_from_slice(&rollup_id.inner)
+            .map_err(Self::Error::incorrect_rollup_id_length)?;
```

Comment:
```suggestion
        let rollup_id = RollupId::try_from_raw(&rollup_id)
            .map_err(Self::Error::incorrect_rollup_id_length)?;
```

rely on the `RollupId` existing implementation